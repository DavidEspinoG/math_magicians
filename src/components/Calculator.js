import '../styles/calculator.css';
import { useState } from 'react';
import CalculatorInput from './CalculatorInput';
import CalculatorButton from './CalculatorButton';
import calculate from '../logic/calculate';

const getColor = (string) => {
  const orangeElements = ['รท', 'x', '-', '+', '='];
  const index = orangeElements.findIndex((element) => element === string);
  if (index !== -1) {
    return 'orange';
  }
  return 'gray';
};

const onClickHandler = (state, buttonName, setState) => {
  setState(calculate(state, buttonName));
};

const Calculator = () => {
  const initialState = {
    total: '0',
    next: null,
    operation: null,
  };
  const [state, setState] = useState(initialState);

  const buttonValues = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x',
    '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div data-testid="calculator" className="calculator">
      <CalculatorInput state={state} />
      <div className="calculator-buttons">
        {buttonValues.map((element) => (
          <CalculatorButton
            key={element}
            content={element}
            color={getColor(element)}
            onClick={onClickHandler}
            state={state}
            setState={setState}
          />
        ))}
      </div>
    </div>
  );
};

export default Calculator;
