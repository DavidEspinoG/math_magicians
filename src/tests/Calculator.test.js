import { render, screen } from "@testing-library/react";
import Calculator from "../components/Calculator";

test('Calculator is on the DOM ', () => {
  render(<Calculator />)
  const calculator = screen.getByTestId('calculator')
  expect(calculator).toBeInTheDocument();
})

test('Match snapshot ', () => {
  const element = render(<Calculator />)
  expect(element).toMatchSnapshot();
})