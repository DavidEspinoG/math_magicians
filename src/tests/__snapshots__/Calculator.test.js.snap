// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator"
      >
        <div
          class="calculator-input"
        >
          <span>
            0
          </span>
        </div>
        <div
          class="calculator-buttons"
        >
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              AC
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              +/-
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              %
            </span>
          </div>
          <div
            class="calculator-button orange "
            role="button"
            tabindex="0"
          >
            <span>
              รท
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              7
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              8
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              9
            </span>
          </div>
          <div
            class="calculator-button orange "
            role="button"
            tabindex="0"
          >
            <span>
              x
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              4
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              5
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              6
            </span>
          </div>
          <div
            class="calculator-button orange "
            role="button"
            tabindex="0"
          >
            <span>
              -
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              1
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              2
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              3
            </span>
          </div>
          <div
            class="calculator-button orange "
            role="button"
            tabindex="0"
          >
            <span>
              +
            </span>
          </div>
          <div
            class="calculator-button gray large"
            role="button"
            tabindex="0"
          >
            <span>
              0
            </span>
          </div>
          <div
            class="calculator-button gray "
            role="button"
            tabindex="0"
          >
            <span>
              .
            </span>
          </div>
          <div
            class="calculator-button orange "
            role="button"
            tabindex="0"
          >
            <span>
              =
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator"
    >
      <div
        class="calculator-input"
      >
        <span>
          0
        </span>
      </div>
      <div
        class="calculator-buttons"
      >
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            AC
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            +/-
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            %
          </span>
        </div>
        <div
          class="calculator-button orange "
          role="button"
          tabindex="0"
        >
          <span>
            รท
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            7
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            8
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            9
          </span>
        </div>
        <div
          class="calculator-button orange "
          role="button"
          tabindex="0"
        >
          <span>
            x
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            4
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            5
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            6
          </span>
        </div>
        <div
          class="calculator-button orange "
          role="button"
          tabindex="0"
        >
          <span>
            -
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            1
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            2
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            3
          </span>
        </div>
        <div
          class="calculator-button orange "
          role="button"
          tabindex="0"
        >
          <span>
            +
          </span>
        </div>
        <div
          class="calculator-button gray large"
          role="button"
          tabindex="0"
        >
          <span>
            0
          </span>
        </div>
        <div
          class="calculator-button gray "
          role="button"
          tabindex="0"
        >
          <span>
            .
          </span>
        </div>
        <div
          class="calculator-button orange "
          role="button"
          tabindex="0"
        >
          <span>
            =
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
